default.nix

```diff
@@ -129,6 +129,78 @@ let
+    # the supplied manifest-updating scripts.
+    , imageTag ? "latest"
+    , os ? "linux"
+    , arch ? pkgs.go.GOARCH
+    , tlsVerify ? true
+    , registryUrl ? "registry.hub.docker.com"
+    , meta ? {}
+    }: let
+      manifest = l.fromJSON (l.readFile imageManifest);
+
+      buildImageBlob = digest:
+        let
+          blobUrl = "https://${registryUrl}/v2/${imageName}/blobs/${digest}";
+          plainDigest = l.replaceStrings ["sha256:"] [""] digest;
+          insecureFlag = l.strings.optionalString (!tlsVerify) "--insecure";
+        in pkgs.runCommand plainDigest {
+          outputHash = plainDigest;
+          outputHashMode = "flat";
+          outputHashAlgo = "sha256";
+        } ''
+          SSL_CERT_FILE="${pkgs.cacert.out}/etc/ssl/certs/ca-bundle.crt";
+
+          # This initial access is expected to fail as we don't have a token.
+          ${pkgs.curl}/bin/curl ${insecureFlag} "${blobUrl}" --head --silent --write-out '%header{www-authenticate}' --output /dev/null > bearer.txt
+          tokenUrl=$(sed -n 's/Bearer realm="\(.*\)",service="\(.*\)",scope="\(.*\)"/\1?service=\2\&scope=\3/p' bearer.txt)
+
+          echo "Token URL: $tokenUrl"
+          ${pkgs.curl}/bin/curl ${insecureFlag} --fail --silent "$tokenUrl" --output token.json
+          token="$(${pkgs.jq}/bin/jq --raw-output .token token.json)"
+
+          echo "Blob URL: ${blobUrl}"
+          ${pkgs.curl}/bin/curl ${insecureFlag} --fail -H "Authorization: Bearer $token" "${blobUrl}" --location --output $out
+        '';
+
+      # Pull the blobs (archives) for all layers, as well as the one for the image's config JSON.
+      layerBlobs = map (layerManifest: buildImageBlob layerManifest.digest) manifest.layers;
+      configBlob = buildImageBlob manifest.config.digest;
+
+      # Write the blob map out to a JSON file for the GO executable to consume.
+      blobMap = l.listToAttrs(map (drv: { name = drv.name; value = drv; }) (layerBlobs ++ [configBlob]));
+      blobMapFile = pkgs.writeText "${imageName}-blobs.json" (l.toJSON blobMap);
+
+      # Convenience scripts for manifest-updating.
+      filter = ''.manifests[] | select((.platform.os=="${os}") and (.platform.architecture=="${arch}")) | .digest'';
+      getManifest = pkgs.writeShellApplication {
+        name = "get-manifest";
+        runtimeInputs = [ pkgs.jq skopeo-nix2container ];
+        text = ''
+          set -e
+          manifest=$(skopeo inspect docker://${registryUrl}/${imageName}:${imageTag} --raw | jq)
+          if echo "$manifest" | jq -e .manifests >/dev/null; then
+            # Multi-arch image, pick the one that matches the supplied platform details.
+            hash=$(echo "$manifest" | jq -r '${filter}')
+            skopeo inspect "docker://${registryUrl}/${imageName}@$hash" --raw | jq
+          else
+            # Single-arch image, return the initial response.
+            echo "$manifest"
+          fi
```

Comment:
This branch is unfortunately necessary, see: https://github.com/containers/skopeo/issues/1283

I've manually validated both cases, but I don't know what would be the best way to check them in CI— happy to take suggestions on that.

Reply:
I don't have a good idea to run this test in the current project test infrastructure. 

If  we want to avoid Internet calls during the test, this means we should run a registry. However, that's not easy in the project CI current context: we don't have a way to properly handle lifecycle of processes and we can't run VMs (to start the registry).

So, maybe we should just accept to call Internet during the test. If the test fails too often, we could then remove it of invest time to improve it. This fragile test could be easily added in the `tests/default.nix` file.

Reply:
I've added a new examples file with three cases, and added a mapAttrs that turns each one into its own test.